web3.js/src/connection.ts

```diff
@@ -3566,7 +3566,16 @@ export class Connection {
       if ('error' in res) {
         throw new Error('failed to get transactions: ' + res.error.message);
       }
-      return res.result;
+      const result = res.result;
+      if (!result) return result;
+
+      return {
+        ...result,
+        transaction: {
+          ...result.transaction,
+          message: new Message(result.transaction.message),
+        },
+      };
```

Comment:
_Slight_ preference for pulling this out into a method (getTransactionResponseFromGetTransactionsResult()` so that `getTransaction` and `getTransactions` can share it, but no big deal – what you've done here is more readable.